components/ws-manager/pkg/manager/monitor.go

```diff
@@ -894,6 +790,31 @@ func (m *Monitor) finalizeWorkspaceContent(ctx context.Context, wso *workspaceOb
 		m.finalizerMap[workspaceID] = cancelReq
 		m.finalizerMapLock.Unlock()
 
+		tpe, err := wso.WorkspaceType()
+		if err != nil {
+			tracing.LogError(span, err)
+			log.WithError(err).Warn("cannot determine workspace type - assuming this is a regular")
+			tpe = api.WorkspaceType_REGULAR
+		}
+
+		if tpe == api.WorkspaceType_PREBUILD {
```

Comment:
relying on the current `doBackup` mechanism breaks if there's no hostIP because of the early exit in line 779. We could add an exception to line 776 akin to
```
doSnapshot := tpe == api.WorkspaceType_PREBUILD
if !doBackup && !doSnapshot && wso.HostIP() == "" {
```

Reply:
>  if there's no hostIP because of the early exit in line 779

But isn't this exactly what we want? If there is not HostIP we cannot create a snapshot successfully. What am I overlooking?

Reply:
We discussed offline, the changes are applied.